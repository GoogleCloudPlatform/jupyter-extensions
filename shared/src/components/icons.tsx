/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { withStyles } from '@material-ui/core';
import {
  Check,
  CheckCircle,
  Close,
  CancelRounded,
  Refresh,
  Info,
  Error,
  Launch,
  MoreVert,
} from '@material-ui/icons';
import BlockIcon from '@material-ui/icons/Block';
import PauseCircleFilledIcon from '@material-ui/icons/PauseCircleOutline';
import HelpIcon from '@material-ui/icons/Help';
import { COLORS } from '../styles';

/** Green check icon */
export const GreenCheck = withStyles({
  root: {
    color: COLORS.green,
    fontSize: '16px',
  },
})(Check);

/** Green check circle icon */
export const GreenCheckCircle = withStyles({
  root: {
    color: COLORS.green,
    fontSize: '16px',
  },
})(CheckCircle);

/** Pause circle icon */
export const PausedCircle = withStyles({
  root: {
    color: COLORS.base,
    fontSize: '16px',
  },
})(PauseCircleFilledIcon);

/** Unknown circle icon */
export const UnknownCircle = withStyles({
  root: {
    color: COLORS.base,
    fontSize: '16px',
  },
})(HelpIcon);

/** Disable icon */
export const GrayDisabled = withStyles({
  root: {
    color: COLORS.base,
    fontSize: '16px',
  },
})(BlockIcon);

/** Red 'X' icon */
export const RedClose = withStyles({
  root: {
    color: COLORS.red,
    fontSize: '16px',
  },
})(Close);

/** Red 'X' Circle icon */
export const RedCloseCircle = withStyles({
  root: {
    color: COLORS.red,
    fontSize: '16px',
  },
})(CancelRounded);

/** Gray pending icon */
export const GrayPending = withStyles({
  root: {
    color: COLORS.base,
    fontSize: '16px',
  },
})(Refresh);

/** Blue information icon */
export const BlueInfo = withStyles({
  root: {
    color: COLORS.link,
    fontSize: '16px',
  },
})(Info);

/** Red information icon */
export const RedError = withStyles({
  root: {
    color: COLORS.red,
    fontSize: '16px',
  },
})(Error);

// tslint:disable-next-line:enforce-name-casing
/** External link Launch icon */
export const SmallLaunchIcon = withStyles({
  root: {
    fontSize: '16px',
    paddingLeft: '2px',
  },
})(Launch);

export const RefreshIcon = withStyles({
  root: {
    fontSize: '16px',
  },
})(Refresh);

export const MenuIcon = withStyles({
  root: {
    fontSize: '20px',
  },
})(MoreVert);
