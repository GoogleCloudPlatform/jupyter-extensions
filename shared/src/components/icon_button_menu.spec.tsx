/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Menu, MenuItem } from '@material-ui/core';
import { shallow } from 'enzyme';
import * as React from 'react';

import {
  IconButtonMenu,
  MenuCloseHandler,
  SmallButton,
} from './icon_button_menu';
import { Add } from '@material-ui/icons';

describe('IconButtonMenu', () => {
  const menuItemsProp = (closeHandler: MenuCloseHandler) => (
    <React.Fragment>
      <MenuItem onClick={closeHandler}>Item 1</MenuItem>
      <MenuItem onClick={closeHandler}>Item 2</MenuItem>
      <MenuItem onClick={closeHandler}>Item 3</MenuItem>
    </React.Fragment>
  );

  it('Renders menu items', () => {
    const iconButtonMenu = shallow(
      <IconButtonMenu menuItems={menuItemsProp} />
    );

    expect(iconButtonMenu).toMatchSnapshot();
  });

  it('Renders with provided icon', () => {
    const iconButtonMenu = shallow(
      <IconButtonMenu menuItems={menuItemsProp} icon={<Add />} />
    );

    expect(iconButtonMenu).toMatchSnapshot();
  });

  it('Opens from icon button and closes when an item is clicked', () => {
    const iconButtonMenu = shallow(
      <IconButtonMenu menuItems={menuItemsProp} />
    );
    expect(iconButtonMenu.find(Menu).prop('open')).toBe(false);

    const openMenuButton = iconButtonMenu.find(SmallButton);
    openMenuButton.simulate('click', {
      currentTarget: openMenuButton.getElement(),
    });

    expect(iconButtonMenu.find(Menu).prop('open')).toBe(true);
    const menuItems = iconButtonMenu.find(MenuItem);
    expect(menuItems.length).toBe(3);
    menuItems.first().simulate('click');
    expect(iconButtonMenu.find(Menu).prop('open')).toBe(false);
  });
});
