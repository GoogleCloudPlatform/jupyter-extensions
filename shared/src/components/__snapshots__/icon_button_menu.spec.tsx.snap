// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonMenu Renders menu items 1`] = `
<span>
  <WithStyles(WithStyles(ForwardRef(IconButton)))
    onClick={[Function]}
  >
    <MoreVertIcon />
  </WithStyles(WithStyles(ForwardRef(IconButton)))>
  <WithStyles(ForwardRef(Menu))
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      Item 1
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      Item 2
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      Item 3
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</span>
`;

exports[`IconButtonMenu Renders with provided icon 1`] = `
<span>
  <WithStyles(WithStyles(ForwardRef(IconButton)))
    onClick={[Function]}
  >
    <AddIcon />
  </WithStyles(WithStyles(ForwardRef(IconButton)))>
  <WithStyles(ForwardRef(Menu))
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      Item 1
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      Item 2
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      Item 3
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</span>
`;
